pennylane/ops/identity.py

```diff
@@ -49,8 +49,9 @@ def label(self, decimals=None, base_label=None):
     def _eigvals(cls, *params):
         return cls.eigvals
 
-    @classmethod
-    def _matrix(cls, *params):
+    # pylint:disable=unused-argument
```

Comment:
I am unsure about this pattern, but I don't know how else to automate things

@albi3ro, shall I use `params, hyperparams` instead? The reason for the star is that I thought if a subclass overwrites this method with a different signature like `(eigvals, num_wires, basis_state)`, the highest-order method  `Operator.compute_matrix` will also work with this input because it does not specify a number of arguments.

Reply:
I am in favour of `*params, **hyperparams`. We decided to do `qnode.construct(args, kwargs)` ages ago, and I really regret it --- it has led to a lot of bugs due to people forgetting to convert between the two. Plus, subclassing, as you say